sdk/identity/identity/src/credentials/visualStudioCodeCredential.ts

```diff
@@ -181,7 +186,11 @@ export class VisualStudioCodeCredential implements TokenCredential {
 
     if (findCredentials === undefined) {
       throw new CredentialUnavailableError(
-        "No implementation of VisualStudioCodeCredential is available (do you need to install and use the `@azure/identity-vscode` extension package?)"
+        [
+          "No implementation of VisualStudioCodeCredential is available.",
+          "You must install the identity-vscode plugin package (`npm install --save-dev @azure/identity-vscode`)",
+          "and enable it by calling `useIdentityPlugin(vsCodePlugin)` before creating a VisualStudioCodeCredential."
+        ].join(" ")
```

Comment:
nit: Seems wasteful to break the string into an array and join it. We can use string literals and new lines to make it as readable.

Reply:
I don’t really agree. It’s joined by spaces, not new lines (in general I think error messages with newlines in them format poorly because you don’t know how wide the user’s console is). This pattern of joining a string array is more of a code formatting concern, and it doesn’t cost more than a few bytes of code size for what I think is much nicer readability, since it won’t execute unless the exception is thrown.

Reply:
Alright! convinced!